### ISSUE TITLE
SSL alert number 40 while handshaking with certain sites

### ISSUE THREAD

**Issue by drook** [openssl/openssl#19723]
**Env:**
OS: FreeBSD 13.1-RELEASE
Openssl: 3.0.5 from ports
Target Site: https://bki-file.equifax.ru

**Issue description:**
When using s_client towards target site, openssl shows minor error (but I'm trying to build an nginx proxy, and it seems to just hate this error). The site clearly requires client certificate (and I have one, but, for NDA reasons, just cannot attach it here) but anyway the error is the same with or without it:

````
/usr/local/bin/openssl s_client -verify_quiet -connect bki-file.equifax.ru:443 -servername bki-file.equifax.ru
CONNECTED(00000004)
0020010108000000:error:0A000410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure:ssl/record/rec_layer_s3.c:1584:SSL alert number 40
---
Certificate chain
0 s:C = RU, ST = \D0\9C\D0\BE\D1\81\D0\BA\D0\B2\D0\B0, L = \D0\9C\D0\BE\D1\81\D0\BA\D0\B2\D0\B0, O = \D0\9E\D0\9E\D0\9E \D0\91\D0\9A\D0\98 \D0\AD\D0\9A\D0\92\D0\98\D0\A4\D0\90\D0\9A\D0\A1, CN = *.equifax.ru
i:C = BE, O = GlobalSign nv-sa, CN = GlobalSign RSA OV SSL CA 2018
a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
v:NotBefore: May 11 13:41:41 2022 GMT; NotAfter: May  8 20:59:59 2023 GMT
1 s:C = BE, O = GlobalSign nv-sa, CN = GlobalSign RSA OV SSL CA 2018
i:OU = GlobalSign Root CA - R3, O = GlobalSign, CN = GlobalSign
a:PKEY: rsaEncryption, 2048 (bit); sigalg: RSA-SHA256
v:NotBefore: Nov 21 00:00:00 2018 GMT; NotAfter: Nov 21 00:00:00 2028 GMT
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIGtDCCBZygAwIBAgIMPLlvc7v+vH6Y2l6xMA0GCSqGSIb3DQEBCwUAMFAxCzAJ
BgNVBAYTAkJFMRkwFwYDVQQKExBHbG9iYWxTaWduIG52LXNhMSYwJAYDVQQDEx1H
bG9iYWxTaWduIFJTQSBPViBTU0wgQ0EgMjAxODAeFw0yMjA1MTExMzQxNDFaFw0y
MzA1MDgyMDU5NTlaMHsxCzAJBgNVBAYTAlJVMRUwEwYDVQQIDAzQnNC+0YHQutCy
0LAxFTATBgNVBAcMDNCc0L7RgdC60LLQsDEnMCUGA1UECgwe0J7QntCeINCR0JrQ
mCDQrdCa0JLQmNCk0JDQmtChMRUwEwYDVQQDDAwqLmVxdWlmYXgucnUwggEiMA0G
CSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDF+4oZ/bEZDfIMCohlfpmNznXAgrmk
jJmvNFcezqGeElnYAUdIETI49h0VTLN0Hu7uTlMcW5nC5g43hNL01EOvrktMT3gF
015/HW15s0s0wgrM10UjV/bIvpKFD1Iu3LfCRJK0BaSGW+6vQ0nSf7cVsHIHJCVW
LbgJ73eTSlLVMPzQu35Qs78SY7z8n241yxJLhREdr4QEkX31l1tNP1iGq5sTK/ru
DwlXrN7pIxhuEJGIjqStxrpIkQsDi9xi3Z1lzv+zNxD4KcAqemidfGlD1kLRs0m0
Rs6kFZNtCWA9o/2mxmFHFlcPjtqffmS+2Ykj2m2+f03I/mQtmR8rNDLZAgMBAAGj
ggNhMIIDXTAOBgNVHQ8BAf8EBAMCBaAwgY4GCCsGAQUFBwEBBIGBMH8wRAYIKwYB
BQUHMAKGOGh0dHA6Ly9zZWN1cmUuZ2xvYmFsc2lnbi5jb20vY2FjZXJ0L2dzcnNh
b3Zzc2xjYTIwMTguY3J0MDcGCCsGAQUFBzABhitodHRwOi8vb2NzcC5nbG9iYWxz
aWduLmNvbS9nc3JzYW92c3NsY2EyMDE4MFYGA1UdIARPME0wQQYJKwYBBAGgMgEU
MDQwMgYIKwYBBQUHAgEWJmh0dHBzOi8vd3d3Lmdsb2JhbHNpZ24uY29tL3JlcG9z
aXRvcnkvMAgGBmeBDAECAjAJBgNVHRMEAjAAMD8GA1UdHwQ4MDYwNKAyoDCGLmh0
dHA6Ly9jcmwuZ2xvYmFsc2lnbi5jb20vZ3Nyc2FvdnNzbGNhMjAxOC5jcmwwMwYD
VR0RBCwwKoIMKi5lcXVpZmF4LnJ1gg53d3cuZXF1aWZheC5ydYIKZXF1aWZheC5y
dTAdBgNVHSUEFjAUBggrBgEFBQcDAQYIKwYBBQUHAwIwHwYDVR0jBBgwFoAU+O9/
8s14Z6jeb48kjYjxhwMCs+swHQYDVR0OBBYEFBuLifFIxHizMYQR0mOxtJH0yYLt
MIIBgAYKKwYBBAHWeQIEAgSCAXAEggFsAWoAdwDoPtDaPvUGNTLnVyi8iWvJA9PL
0RFr7Otp4Xd9bQa9bgAAAYCzWqaCAAAEAwBIMEYCIQC0fBaSr/MexbBhFavxOSjR
g8wPZCIrFhFFslfhDmtAUQIhAKeENGyVZKJgup9XUye2C2aD4xxaw8XuTAQunXju
8h2mAHYAb1N2rDHwMRnYmQCkURX/dxUcEdkCwQApBo2yCJo32RMAAAGAs1qkvAAA
BAMARzBFAiBltkgzhpSA9kkMclzIySBDw/9BBBtRp3me4Q7tdDWK3AIhAKEtr6LL
L8IpuLzvV+AQl9JZAY9exMXdD9SB+bBxl/25AHcAs3N3B+GEUPhjhtYFqdwRCUp5
LbFnDAuH3PADDnk2pZoAAAGAs1qmAQAABAMASDBGAiEAmjgVP4s50PufaD0UphbN
x9f87hbenhwsjjoC4cf2kO0CIQD7LqMoISG1EZm3Lz9YK/VpgoxyqDdDGRz8m/pE
32Zz7TANBgkqhkiG9w0BAQsFAAOCAQEAJkm4+m7SY5o/lrAKMFJvUujRLsD5T8xQ
u8zWs9jPkhgqdxTRN3BOIRJovc5huDWf9fJ3JTkGOOuDg8vW63gFr1aBAb6TDq5B
xN9saWibAopA1HQyEy8ySSBHhfNvvIFzhQlb3uprJmSLg2jNRpycB0t6QCvlllbk
dBnmKHzLjmudh6V/mw3W1LwXRfTiNHdRVL2NMKLIMevp5i2ivRLXBhW4RzjxgKnD
daqq82P2Gc1VkTB3QTGbmpZhmjRRxW1xiuO1yJNOER0bPQ8Iss14Pb7mSqwBtFZj
J4kUTjE1a73bY/ZTJUGhGkLWrHVQrwyo+E6rMJUgZU0jNJInbw9tOQ==
-----END CERTIFICATE-----
subject=C = RU, ST = \D0\9C\D0\BE\D1\81\D0\BA\D0\B2\D0\B0, L = \D0\9C\D0\BE\D1\81\D0\BA\D0\B2\D0\B0, O = \D0\9E\D0\9E\D0\9E \D0\91\D0\9A\D0\98 \D0\AD\D0\9A\D0\92\D0\98\D0\A4\D0\90\D0\9A\D0\A1, CN = *.equifax.ru
issuer=C = BE, O = GlobalSign nv-sa, CN = GlobalSign RSA OV SSL CA 2018
---
Acceptable client certificate CA names
C = RU, ST = Russia, O = Equifax Credit Services LLC, CN = Equifax TLS
C = RU, ST = Russia, L = Moscow, O = Equifax Credit Services LLC, CN = Equifax Credit Services CA
Client Certificate Types: RSA sign, DSA sign, ECDSA sign
Requested Signature Algorithms: RSA+SHA512:DSA+SHA512:ECDSA+SHA512:RSA+SHA384:DSA+SHA384:ECDSA+SHA384:RSA+SHA256:DSA+SHA256:ECDSA+SHA256:RSA+SHA224:DSA+SHA224:ECDSA+SHA224:RSA+SHA1:DSA+SHA1:ECDSA+SHA1
Shared Requested Signature Algorithms: RSA+SHA512:DSA+SHA512:ECDSA+SHA512:RSA+SHA384:DSA+SHA384:ECDSA+SHA384:RSA+SHA256:DSA+SHA256:ECDSA+SHA256:RSA+SHA224:DSA+SHA224:ECDSA+SHA224
Peer signing digest: SHA512
Peer signature type: RSA
Server Temp Key: ECDH, prime256v1, 256 bits
---
SSL handshake has read 3539 bytes and written 463 bytes
Verification: OK
---
New, TLSv1.2, Cipher is ECDHE-RSA-AES256-GCM-SHA384
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
Protocol  : TLSv1.2
Cipher    : ECDHE-RSA-AES256-GCM-SHA384
Session-ID:
Session-ID-ctx:
Master-Key: 3D6790AE84E1AD101DEB9BA5AC18FECAE2155667202F7CFF0D7F0C12605DD926FF63A2DBF8EE2D924CE291BDE1FE4B4D
PSK identity: None
PSK identity hint: None
SRP username: None
Start Time: 1668690458
Timeout   : 7200 (sec)
Verify return code: 0 (ok)
Extended master secret: yes
---
````

**Comment by t8m** [comment]
The target server simply sends the alert. There is not much we can do with that. I see the same error on that site with openssl versions up to 1.0.2.

**Comment by mathenry1** [comment]
I'm having the same issue on my website based on Buy Painkiller and Pain Pills from ONLINE MEDS STORE
Offering a wide range of medications including painkiller, CBD products, [buy mushroom online](https://onlinemedsstore.uk/), antidepressants, and cancer-curing medicines.

**Comment by nhorman** [comment]
marking as inactive, will be closed at the completion of 3.4 dev without further input
